src/components/results/securityanalysis/security-analysis-result-utils.ts

```diff
@@ -131,36 +124,130 @@ export const flattenNmKResultsConstraints = (
 };
 
 export const securityAnalysisTableNColumnsDefinition = (
-    intl: IntlShape
+    intl: IntlShape,
+    makeColumn: (customColDef: CustomColDef) => any
 ): ColDef[] => [
-    {
+    makeColumn({
         headerName: intl.formatMessage({ id: 'Equipment' }),
         field: 'subjectId',
-        filter: 'agTextColumnFilter',
-    },
-    {
-        headerName: intl.formatMessage({ id: 'LimitType' }),
+        /*  valueGetter: contingencyGetterValues,
+        cellRenderer: ContingencyCellRenderer, */
+        isSortable: true,
+        isFilterable: true,
+        filterParams: {
+            filterUIType: FILTER_UI_TYPES.TEXT,
+            filterComparators: [
+                FILTER_TEXT_COMPARATORS.STARTS_WITH,
+                FILTER_TEXT_COMPARATORS.CONTAINS,
+            ],
+        },
+    }),
+
+    makeColumn({
+        headerName: intl.formatMessage({ id: 'ViolationType' }),
         field: 'limitType',
-        filter: 'agTextColumnFilter',
-    },
-    {
+        isSortable: true,
+        isFilterable: true,
+    }),
+
+    makeColumn({
+        headerName: intl.formatMessage({ id: 'LimitName' }),
+        field: 'limitName',
+        isSortable: true,
+        isFilterable: true,
+        filterParams: {
+            filterUIType: FILTER_UI_TYPES.TEXT,
+            filterComparators: [
+                FILTER_TEXT_COMPARATORS.STARTS_WITH,
+                FILTER_TEXT_COMPARATORS.CONTAINS,
+            ],
+        },
+    }),
+
+    makeColumn({
         headerName: intl.formatMessage({ id: 'Limit' }),
         field: 'limit',
+        numeric: true,
+        isSortable: true,
+        isFilterable: true,
         valueFormatter: (params: ValueFormatterParams) =>
             params.data?.limit?.toFixed(1),
-    },
-    {
-        headerName: intl.formatMessage({ id: 'Value' }),
+        filterParams: {
+            filterUIType: FILTER_UI_TYPES.NUMBER,
+            filterComparators: [
+                FILTER_NUMBER_COMPARATORS.NOT_EQUAL,
+                FILTER_NUMBER_COMPARATORS.GREATER_THAN_OR_EQUAL,
+                FILTER_NUMBER_COMPARATORS.LESS_THAN_OR_EQUAL,
+            ],
+        },
+    }),
+    makeColumn({
```

Comment:
I could be nice to create a "makeNumberColumn" since they almost all have the same configurations
Same for text columns

Reply:
I prefer having a single method and passing the parameters to it rather than having two methods with the same parameters, as that would be redundant. I did better; I factored makeColumn for n and n-k in security-analysis-result-utils.ts